framework/src/bcs/NodalBC.C

```diff
@@ -43,6 +77,13 @@ NodalBC::computeResidual(NumericVector<Number> & residual)
     dof_id_type & dof_idx = _var.nodalDofIndex();
     _qp = 0;
     residual.set(dof_idx, computeQpResidual());
+
+    if (_has_save_in)
+    {
+      Threads::spin_mutex::scoped_lock lock(Threads::spin_mtx);
+      for (unsigned int i=0; i<_save_in.size(); i++)
+        _save_in[i]->sys().solution().set(dof_idx, 0);
```

Comment:
Yeah - something like that.  Have `NodalBC` call `FEProblem::setNodalBCSaveIn(true)` during the constructor.